src/main/java/org/folio/fqm/service/EntityTypeService.java

```diff
@@ -195,7 +196,10 @@ private ColumnValues getFieldValuesFromApi(Field field, String searchText, List<
     Set<ValueWithLabel> resultSet = new HashSet<>();
     for (String tenantId : tenantsToQuery) {
       try {
-        String rawJson = crossTenantHttpClient.get(field.getValueSourceApi().getPath(), Map.of("limit", String.valueOf(COLUMN_VALUE_DEFAULT_PAGE_SIZE)), tenantId);
+        ValueSourceApi valueSourceApi = field.getValueSourceApi();
+        Map<String, String> queryParams = Optional.ofNullable(valueSourceApi.getQueryParams())
+          .orElseGet(() -> Map.of("limit", String.valueOf(COLUMN_VALUE_DEFAULT_PAGE_SIZE)));
```

Comment:
Slightly more efficient, since it doesn't involve instantiating an `Optional`:
```suggestion
        Map<String, String> queryParams = Objects.requireNonNullElseGet(valueSourceApi.getQueryParams(),
        () -> Map.of("limit", String.valueOf(COLUMN_VALUE_DEFAULT_PAGE_SIZE)));
```

Reply:
I hate the name of that method soooo much... It has "require non-null" in the name, while simultaneously providing a mechanism for dealing with it being null 🤦